viewing_party/party.py

```diff
@@ -1,23 +1,186 @@
 # ------------- WAVE 1 --------------------
 
+from tests.test_constants import GENRE_1, MOVIE_TITLE_1, RATING_1
+
+
 def create_movie(title, genre, rating):
-    pass
+    new_movie = {}
+
+
+    if title and genre and rating:
+        new_movie["title"] = title
+        new_movie["genre"] = genre
+        new_movie["rating"] = rating
+        return new_movie
+    else: 
+        return None
+
+    
+def add_to_watched(user_data, movie):
+    user_data = {"watched": []}
+    
+    movie = {
+        "title": MOVIE_TITLE_1,
+        "genre": GENRE_1,
+        "rating": RATING_1
+    }
+
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data = {"watchlist": []}
+    movie = {
+    "title": MOVIE_TITLE_1,
+    "genre": GENRE_1,
+    "rating": RATING_1
+    }
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+
+    
+    for key, value in user_data.items():
+        if key == "watchlist":
+            for i in value:
+                for key, value in i.items():
+                    if value == title:
+                        user_data["watched"].append(title)
+                        user_data["watchlist"].remove(i)
+        else:
+            return user_data
+        
+    return user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    rating_total = []
 
+    for i in user_data["watched"]:
+        rating_total.append(i["rating"])
+    if len(rating_total) != 0:
+        return sum(rating_total)/len(rating_total)
+    else: 
+        return 0.0
 
+
+def get_most_watched_genre(user_data):
+ 
+    popular_genre = []
+    for i in user_data["watched"]:
+        popular_genre.append(i["genre"])
+    if len(popular_genre) != 0:
+        return max(set(popular_genre), key = popular_genre.count)
+    else: 
+        return None

```

Comment:
Instead of returning None in an else block, which kind of obscures the scenario when None is returned, we can use a guard clause to invert the logic. Doing so calls attention to the fact that we return None when the watched list is empty.

So your refactored function would have 
```python 
if not user_data["watched"]:
    return None

# rest of your logic goes after the guard clause
```